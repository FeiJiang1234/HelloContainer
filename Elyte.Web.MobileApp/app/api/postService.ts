import { CommentModel } from 'el/models/post/commentModel';
import { PageResult } from '../models/pageResult';
import { PostModel } from '../models/post/PostModel';
import { ResponseResult } from '../models/responseResult';
import http from './httpService';

function create(data, file) {
    const formData = new FormData();
    formData.append('details', data.details);
    formData.append('file', file);
    return http.post(`posts`, formData);
}

function getPosts(pageNumber, pageSize) {
    return http.get<null, ResponseResult<PageResult<PostModel>>>(
        `mobile/posts?pageNumber=${pageNumber}&pagesize=${pageSize}`,
    );
}

function getLatestPosts(postsAfter) {
    return http.get<null, ResponseResult<PostModel[]>>(
        `mobile/posts/latest?postsAfter=${postsAfter}`,
    );
}

function addPostComment(data) {
    return http.post(`posts/${data.postId}/comment`, data);
}

function addCommentReply (data) {
    return http.post(`posts/${data.postId}/comments/${data.commentId}/reply`, data);
}

function getPostComments(postId) {
    return http.get<null, ResponseResult<CommentModel[]>>(`posts/${postId}/comments`);
}

function gePost(postId) {
    return http.get(`posts/${postId}`);
}

function likePost(athleteId, postId) {
    return http.post(`posts/${postId}/athletes/${athleteId}/like`, {});
}

function unlikePost(athleteId, postId) {
    return http.delete(`posts/${postId}/athletes/${athleteId}/unlike`, {});
}

function getPostLikes(postId) {
    return http.get(`posts/${postId}/likes`);
}

function favoritePost(athleteId, postId) {
    return http.post(`posts/${postId}/athletes/${athleteId}/favorite`, {});
}

function unFavoritePost(athleteId, postId) {
    return http.delete(`posts/${postId}/athletes/${athleteId}/unFavorite`, {});
}

function sharePost (postId) {
    return http.put(`posts/${postId}/share`);
}

function unsharePost (postId) {
    return http.put(`posts/${postId}/unshare`);
}

function deletePostComment (postId, commentId) {
    return http.delete(`posts/${postId}/comments/${commentId}`, {});
}

function removePost (postId) {
    return http.delete(`posts/${postId}`);
}


function createPostStats (data) {
    return http.post(`posts/stats`, data);
}

export default {
    create,
    getPosts,
    getLatestPosts,
    addPostComment,
    addCommentReply,
    getPostComments,
    gePost,
    likePost,
    unlikePost,
    getPostLikes,
    favoritePost,
    unFavoritePost,
    sharePost,
    unsharePost,
    deletePostComment,
    removePost,
    createPostStats
};
