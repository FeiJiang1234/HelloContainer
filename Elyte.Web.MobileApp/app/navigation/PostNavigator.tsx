import React from 'react';
import { createStackNavigator, TransitionPresets } from '@react-navigation/stack';
import routes from './routes';
import PostScreen from 'el/screen/post/PostScreen';
import PostCreateScreen from 'el/screen/post/PostCreateScreen';
import MyProfileScreen from 'el/screen/accountInfo/MyProfileScreen';
import EditProfileScreen from 'el/screen/accountInfo/EditProfileScreen';
import AccountSecurityScreen from 'el/screen/accountInfo/AccountSecurityScreen';
import PaymentHistoryScreen from 'el/screen/accountInfo/PaymentHistoryScreen';
import ChatUserListScreen from 'el/screen/message/ChatUserListScreen';
import ChatDialogBoxScreen from 'el/screen/message/ChatDialogBoxScreen';
import FindChatUserScreen from 'el/screen/message/FindChatUserScreen';
import TeamInvitesScreen from 'el/screen/accountInfo/TeamInvitesScreen';
import ChangePhoneNumberScreen from 'el/screen/accountInfo/ChangePhoneNumberScreen';
import SearchScreen from 'el/screen/search/SearchScreen';
import NotificationScreen from 'el/screen/notification/NotificationScreen';
import AthleteOfficiateRequestScreen from 'el/screen/accountInfo/AthleteOfficiateRequestScreen';
import AthleteJoinTeamRequestScreen from 'el/screen/accountInfo/AthleteJoinTeamRequestScreen';
import DeleteAccountScreen from 'el/screen/accountInfo/DeleteAccountScreen';

const Stack = createStackNavigator();

const PostNavigator = () => {
    return (
        <Stack.Navigator screenOptions={{ headerShown: false, ...TransitionPresets.SlideFromRightIOS }} initialRouteName={routes.PostList}>
            <Stack.Screen name={routes.PostList} component={PostScreen} />
            <Stack.Screen name={routes.PostCreate} component={PostCreateScreen} />
            <Stack.Screen name={routes.MyProfile} component={MyProfileScreen} />
            <Stack.Screen name={routes.DeleteAccount} component={DeleteAccountScreen} />
            <Stack.Screen name={routes.AthleteOfficiateRequest} component={AthleteOfficiateRequestScreen} />
            <Stack.Screen name={routes.AthleteJoinTeamRequest} component={AthleteJoinTeamRequestScreen} />
            <Stack.Screen name={routes.TeamInvites} component={TeamInvitesScreen} />
            <Stack.Screen name={routes.EditProfile} component={EditProfileScreen} />
            <Stack.Screen name={routes.ChangePhoneNumber} component={ChangePhoneNumberScreen} />
            <Stack.Screen name={routes.AccountSecurity} component={AccountSecurityScreen} />
            <Stack.Screen name={routes.PaymentHistory} component={PaymentHistoryScreen} />
            <Stack.Screen name={routes.ChatUserList} component={ChatUserListScreen} />
            <Stack.Screen name={routes.ChatDialogBox} component={ChatDialogBoxScreen} />
            <Stack.Screen name={routes.FindChatUser} component={FindChatUserScreen} />
            <Stack.Screen name={routes.Search} component={SearchScreen} />
            <Stack.Screen name={routes.Notification} component={NotificationScreen} />
        </Stack.Navigator>
    );
};

export default PostNavigator;
