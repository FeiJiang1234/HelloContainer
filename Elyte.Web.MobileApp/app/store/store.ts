import { configureStore } from '@reduxjs/toolkit';
import athleteReducer from './slices/athleteSlice';
import postReducer from './slices/postSlice';
import requestReducer from './slices/requestSlice';
import routeReducer from './slices/routeSlice';
import tabBarReducer from './slices/tabBarSlice';

export const store = configureStore({
    reducer: {
        athlete: athleteReducer,
        posts: postReducer,
        request: requestReducer,
        route: routeReducer,
        tabBar: tabBarReducer,
    },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
