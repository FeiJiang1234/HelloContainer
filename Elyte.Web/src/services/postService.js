import http from './httpService';

function create (data) {
    const formData = new FormData();
    formData.append('details', data.details);
    formData.append('file', data.file);
    return http.post(`posts`, formData);
}

function gePosts () {
    return http.get(`posts`);
}

function gePost (postId) {
    return http.get(`posts/${postId}`);
}

function removePost (postId) {
    return http.delete(`posts/${postId}`);
}

function sharePost (postId) {
    return http.put(`posts/${postId}/share`);
}

function unsharePost (postId) {
    return http.put(`posts/${postId}/unshare`);
}

function deletePostComment (postId, commentId) {
    return http.delete(`posts/${postId}/comments/${commentId}`, {});
}

function likePost (athleteId, postId) {
    return http.post(`posts/${postId}/athletes/${athleteId}/like`, {});
}

function unlikePost (athleteId, postId) {
    return http.delete(`posts/${postId}/athletes/${athleteId}/unlike`, {});
}

function favoritePost (athleteId, postId) {
    return http.post(`posts/${postId}/athletes/${athleteId}/favorite`, {});
}

function unFavoritePost (athleteId, postId) {
    return http.delete(`posts/${postId}/athletes/${athleteId}/unFavorite`, {});
}

function getPostComments (postId) {
    return http.get(`posts/${postId}/comments`, { hideGlobalLoading: true });
}

function getPostLikes (postId) {
    return http.get(`posts/${postId}/likes`, { hideGlobalLoading: true });
}

function addCommentReply (data) {
    return http.post(`posts/${data.postId}/comments/${data.commentId}/reply`, data, { hideGlobalLoading: true });
}

function addPostComment (data) {
    return http.post(`posts/${data.postId}/comment`, data, { hideGlobalLoading: true });
}


function createPostStats (data) {
    return http.post(`posts/stats`, data);
}

export default {
    create,
    gePosts,
    gePost,
    removePost,
    likePost,
    unlikePost,
    favoritePost,
    unFavoritePost,
    addPostComment,
    getPostComments,
    addCommentReply,
    deletePostComment,
    sharePost,
    unsharePost,
    getPostLikes,
    createPostStats
};
