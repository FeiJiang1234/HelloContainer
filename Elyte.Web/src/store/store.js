import { combineReducers, createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { userReducer } from './reducers';

const reducer = combineReducers({
    user: userReducer,
});

const initialState = {};

const middleware = [thunk];

const store = createStore(reducer, initialState, composeWithDevTools(applyMiddleware(...middleware)));
export default store;
