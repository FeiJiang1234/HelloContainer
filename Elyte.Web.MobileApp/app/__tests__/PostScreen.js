import React from 'react';
import { waitFor } from '@testing-library/react-native';
import postService from '../api/postService';
import PostNavigator from '../navigation/PostNavigator';
import PostCard from '../screen/post/components/PostCard';
import { Provider } from 'react-redux';
import { store } from 'el/store/store';
import { NavigationContainer } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import { NativeBaseProvider } from 'native-base';

const renderRoot = ui => {
    const inset = {
        frame: { x: 0, y: 0, width: 0, height: 0 },
        insets: { top: 0, left: 0, right: 0, bottom: 0 },
    };
    return render(
        <NativeBaseProvider initialWindowMetrics={inset}>
            <NavigationContainer>{ui}</NavigationContainer>
        </NativeBaseProvider>,
    );
};

var moment = require('moment');
jest.mock('../api/postService');

const post1 = {
    postId: '1',
    creatorId: '1',
    creatorName: 'fred',
    creatorPictureUrl: 'http:fred.avatar',
    details: 'Today is Friday',
    createdDate: moment.utc(new Date()).format('YYYY-MM-DD HH:m:s'),
    likeCounts: 3,
    commentCounts: 5,
    comments: [
        {
            comment: 'Hello Friday',
            commenterAvatar: 'http:fred.avatar',
            commenterName: 'fred',
            canDelete: true,
        },
        {
            comment: 'Hello Sunday',
            commenterAvatar: 'http:tom.avatar',
            commenterName: 'tom',
            canDelete: false,
        },
    ],
};

const post2 = {
    postId: '2',
    creatorId: '2',
    creatorName: 'tony',
    creatorPictureUrl: 'http:tony.avatar',
    details: 'I love Friday',
    createdDate: moment.utc(new Date()).format('YYYY-MM-DD HH:m:s'),
    likeCounts: 4,
    commentCounts: 10,
};

const mockPosts = {
    value: {
        items: [post1, post2],
        totalCount: 100,
    },
    code: 200,
    message: null,
    errors: null,
};

test('render post screen', async () => {
    postService.getPosts.mockReturnValue(mockPosts);

    // Arrange
    const { getByText } = renderRoot(
        <Provider store={store}>
            <PostNavigator />
        </Provider>,
    );

    // Assert
    await waitFor(() => {
        getByText('fred');
        getByText('Today is Friday');
        getByText('Hello Friday');
        getByText('3 likes 5 comments');
        getByText('Read all comments (5)');
        getByText('tony');
        getByText('I love Friday');
        getByText('Hello Sunday');
        getByText('4 likes 10 comments');
        getByText('Read all comments (10)');
    });
});

test('render post card', () => {
    // Arrange
    const postCard = renderRoot(
        <Provider store={store}>
            <PostCard {...post1} />
        </Provider>,
    );

    // Assert
    expect(postCard).toMatchSnapshot();
});
