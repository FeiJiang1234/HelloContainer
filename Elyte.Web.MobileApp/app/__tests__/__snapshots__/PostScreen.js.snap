// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render post card 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "borderBottomColor": "#F0F2F7",
        "borderBottomWidth": 1,
        "marginBottom": 16,
        "paddingBottom": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "marginTop": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            {
              "uri": "http:fred.avatar",
            }
          }
          style={
            [
              {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              {},
            ]
          }
        />
        <View
          style={
            {
              "marginLeft": 8,
            }
          }
        >
          <Text
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#171717",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            fred
          </Text>
          <Text
            dataSet={{}}
            style={
              [
                {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                },
                {
                  "color": "#808A9E",
                },
              ]
            }
          >
            a few seconds ago
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={23}
          bbWidth={6}
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "height": 23,
                "width": 6,
              },
              {
                "flex": 0,
                "height": 23,
                "width": 6,
              },
            ]
          }
          vbHeight={23}
          vbWidth={6}
          xml="
<svg viewBox="0 0 6 23" fill="none" xmlns="http://www.w3.org/2000/svg">
  <circle cx="2.91" cy="2.91" r="1.91" stroke="#C0C5D0" stroke-width="1.91"/>
  <circle cx="2.91" cy="11.4998" r="1.91" stroke="#C0C5D0" stroke-width="1.91"/>
  <circle cx="2.91" cy="20.0902" r="1.91" stroke="#C0C5D0" stroke-width="1.91"/>
</svg>

"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGCircle
              cx={2.91}
              cy={2.91}
              propList={
                [
                  "stroke",
                  "strokeWidth",
                ]
              }
              r={1.91}
              stroke={4290823632}
              strokeWidth={1.91}
            />
            <RNSVGCircle
              cx={2.91}
              cy={11.4998}
              propList={
                [
                  "stroke",
                  "strokeWidth",
                ]
              }
              r={1.91}
              stroke={4290823632}
              strokeWidth={1.91}
            />
            <RNSVGCircle
              cx={2.91}
              cy={20.0902}
              propList={
                [
                  "stroke",
                  "strokeWidth",
                ]
              }
              r={1.91}
              stroke={4290823632}
              strokeWidth={1.91}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      dataSet={{}}
      style={
        {
          "marginBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        Today is Friday
      </Text>
    </View>
    <View
      dataSet={{}}
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <View
        dataSet={{}}
        style={
          {
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={17}
          bbWidth={21}
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#C0C5D0"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "height": 17,
                "width": 21,
              },
              {
                "flex": 0,
                "height": 17,
                "width": 21,
              },
            ]
          }
          vbHeight={17}
          vbWidth={21}
          xml="
<svg viewBox="0 0 21 17" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M14.3324 1.00001C12.6525 1.00677 11.0882 1.85617 10.1676 3.26136C8.94886 1.43013 6.67694 0.609834 4.56943 1.24008C2.46192 1.87032 1.01334 3.80323 1 6.00292C1 13.5029 10.1676 16 10.1676 16C10.1676 16 19.3353 13.5029 19.3353 6.00292C19.3376 4.67535 18.8112 3.40149 17.8725 2.46276C16.9338 1.52403 15.6599 0.997685 14.3324 1.00001Z" stroke-width="1.91"/>
<path d="M14.3323 4.33547C14.7753 4.33313 15.2009 4.50808 15.5141 4.82133C15.8274 5.13458 16.0023 5.56011 16 6.00311C15.975 7.46874 15.3614 8.86269 14.2974 9.87098" stroke-width="1.91"/>
</svg>

"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={4290823632}
          >
            <RNSVGPath
              clipRule={0}
              d="M14.3324 1.00001C12.6525 1.00677 11.0882 1.85617 10.1676 3.26136C8.94886 1.43013 6.67694 0.609834 4.56943 1.24008C2.46192 1.87032 1.01334 3.80323 1 6.00292C1 13.5029 10.1676 16 10.1676 16C10.1676 16 19.3353 13.5029 19.3353 6.00292C19.3376 4.67535 18.8112 3.40149 17.8725 2.46276C16.9338 1.52403 15.6599 0.997685 14.3324 1.00001Z"
              fillRule={0}
              propList={
                [
                  "fillRule",
                  "strokeWidth",
                ]
              }
              strokeWidth={1.91}
            />
            <RNSVGPath
              d="M14.3323 4.33547C14.7753 4.33313 15.2009 4.50808 15.5141 4.82133C15.8274 5.13458 16.0023 5.56011 16 6.00311C15.975 7.46874 15.3614 8.86269 14.2974 9.87098"
              propList={
                [
                  "strokeWidth",
                ]
              }
              strokeWidth={1.91}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="none"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "height": 20,
                  "marginLeft": 16,
                  "width": 20,
                },
                {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            vbHeight={20}
            vbWidth={20}
            xml="
<svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M15.5739 6.66687V13.1479C15.5739 14.4878 14.4876 15.5741 13.1477 15.5741H11.5274V18.0002L6.66667 15.5741H3.42615C2.78269 15.5741 2.16559 15.3184 1.7106 14.8635C1.25561 14.4085 1 13.7914 1 13.1479V6.66687C1 5.32695 2.08622 4.24072 3.42615 4.24072H13.1477C14.4876 4.24072 15.5739 5.32695 15.5739 6.66687Z" stroke="#C0C5D0" stroke-width="1.91"/>
<path d="M15.5738 12.3333H16.4221C17.7422 12.3149 18.8053 11.2443 18.8144 9.92415V3.42615C18.8144 2.08622 17.7281 1 16.3882 1H6.66666C5.32673 1 4.24051 2.08622 4.24051 3.42615V4.27445" stroke="#C0C5D0" stroke-width="1.91"/>
<path d="M4.24051 9.90706H5.86077" stroke="#C0C5D0" stroke-width="1.91"/>
<path d="M7.48105 9.90706H9.10131" stroke="#C0C5D0" stroke-width="1.91"/>
<path d="M10.7131 9.90706H12.3333" stroke="#C0C5D0" stroke-width="1.91"/>
</svg>

"
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                clipRule={0}
                d="M15.5739 6.66687V13.1479C15.5739 14.4878 14.4876 15.5741 13.1477 15.5741H11.5274V18.0002L6.66667 15.5741H3.42615C2.78269 15.5741 2.16559 15.3184 1.7106 14.8635C1.25561 14.4085 1 13.7914 1 13.1479V6.66687C1 5.32695 2.08622 4.24072 3.42615 4.24072H13.1477C14.4876 4.24072 15.5739 5.32695 15.5739 6.66687Z"
                fillRule={0}
                propList={
                  [
                    "fillRule",
                    "stroke",
                    "strokeWidth",
                  ]
                }
                stroke={4290823632}
                strokeWidth={1.91}
              />
              <RNSVGPath
                d="M15.5738 12.3333H16.4221C17.7422 12.3149 18.8053 11.2443 18.8144 9.92415V3.42615C18.8144 2.08622 17.7281 1 16.3882 1H6.66666C5.32673 1 4.24051 2.08622 4.24051 3.42615V4.27445"
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                  ]
                }
                stroke={4290823632}
                strokeWidth={1.91}
              />
              <RNSVGPath
                d="M4.24051 9.90706H5.86077"
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                  ]
                }
                stroke={4290823632}
                strokeWidth={1.91}
              />
              <RNSVGPath
                d="M7.48105 9.90706H9.10131"
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                  ]
                }
                stroke={4290823632}
                strokeWidth={1.91}
              />
              <RNSVGPath
                d="M10.7131 9.90706H12.3333"
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                  ]
                }
                stroke={4290823632}
                strokeWidth={1.91}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={20}
          bbWidth={15}
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#C0C5D0"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "height": 20,
                "marginLeft": 16,
                "width": 15,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 15,
              },
            ]
          }
          vbHeight={20}
          vbWidth={15}
          xml="
<svg viewBox="0 0 15 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M13.954 18L7.477 14.7615L1 18V1H13.954V18Z" stroke-width="1.9"/>
</svg>


"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
              ]
            }
            stroke={4290823632}
          >
            <RNSVGPath
              clipRule={0}
              d="M13.954 18L7.477 14.7615L1 18V1H13.954V18Z"
              fillRule={0}
              propList={
                [
                  "fillRule",
                  "strokeWidth",
                ]
              }
              strokeWidth={1.9}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          dataSet={{}}
          style={
            [
              {
                "backgroundColor": undefined,
                "color": "#171717",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              },
              {
                "color": "#808A9E",
                "fontSize": 12,
                "marginLeft": 8,
              },
            ]
          }
        >
          3
           likes 
          5
           comments
        </Text>
      </View>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={17}
        bbWidth={21}
        fill="none"
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "height": 17,
              "width": 21,
            },
            {
              "flex": 0,
              "height": 17,
              "width": 21,
            },
          ]
        }
        vbHeight={17}
        vbWidth={17}
        xml="
<svg viewBox="0 0 17 17" fill="none" xmlns="http://www.w3.org/2000/svg">
  <circle cx="13.9571" cy="3.04286" r="2.04286" stroke="#C0C5D0" stroke-width="1.91"/>
  <circle cx="3.04286" cy="8.49989" r="2.04286" stroke="#C0C5D0" stroke-width="1.91"/>
  <circle cx="13.9571" cy="13.9569" r="2.04286" stroke="#C0C5D0" stroke-width="1.91"/>
  <path d="M12.1286 3.95703L4.87143 7.5856" stroke="#C0C5D0" stroke-width="1.91"/>
  <path d="M12.1286 13.0426L4.87143 9.41406" stroke="#C0C5D0" stroke-width="1.91"/>
</svg>
"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx={13.9571}
            cy={3.04286}
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            r={2.04286}
            stroke={4290823632}
            strokeWidth={1.91}
          />
          <RNSVGCircle
            cx={3.04286}
            cy={8.49989}
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            r={2.04286}
            stroke={4290823632}
            strokeWidth={1.91}
          />
          <RNSVGCircle
            cx={13.9571}
            cy={13.9569}
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            r={2.04286}
            stroke={4290823632}
            strokeWidth={1.91}
          />
          <RNSVGPath
            d="M12.1286 3.95703L4.87143 7.5856"
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={4290823632}
            strokeWidth={1.91}
          />
          <RNSVGPath
            d="M12.1286 13.0426L4.87143 9.41406"
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={4290823632}
            strokeWidth={1.91}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#17C476",
            "fontFamily": undefined,
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 18,
            "marginBottom": 8,
            "marginLeft": 16,
            "marginTop": 8,
            "textDecorationLine": undefined,
          }
        }
      >
        Read all comments (
        5
        )
      </Text>
    </View>
    <View
      dataSet={{}}
      style={{}}
    >
      <View
        dataSet={{}}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 4,
            "marginLeft": 16,
          }
        }
      >
        <Image
          source={
            {
              "uri": "http:fred.avatar",
            }
          }
          style={
            [
              {
                "borderRadius": 14,
                "height": 28,
                "width": 28,
              },
              {},
            ]
          }
        />
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "marginLeft": 4,
              "marginRight": 4,
              "textDecorationLine": undefined,
            }
          }
        >
          fred
          :
        </Text>
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#808A9E",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Hello Friday
        </Text>
      </View>
      <View
        dataSet={{}}
        style={
          {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginBottom": 4,
            "marginRight": 16,
          }
        }
      >
        <Text
          allowFontScaling={false}
          dataSet={{}}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#E0E0E0",
                "fontSize": 24,
                "height": 24,
                "lineHeight": 24,
                "width": 24,
              },
              {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="reply"
        >
          󰑚
        </Text>
        <Text
          allowFontScaling={false}
          dataSet={{}}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#E0E0E0",
                "fontSize": 24,
                "height": 24,
                "lineHeight": 24,
                "marginLeft": 8,
                "width": 24,
              },
              {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="delete"
        >
          󰆴
        </Text>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    >
      <View
        dataSet={{}}
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 4,
            "marginLeft": 16,
          }
        }
      >
        <Image
          source={
            {
              "uri": "http:tom.avatar",
            }
          }
          style={
            [
              {
                "borderRadius": 14,
                "height": 28,
                "width": 28,
              },
              {},
            ]
          }
        />
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "marginLeft": 4,
              "marginRight": 4,
              "textDecorationLine": undefined,
            }
          }
        >
          tom
          :
        </Text>
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#808A9E",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Hello Sunday
        </Text>
      </View>
      <View
        dataSet={{}}
        style={
          {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginBottom": 4,
            "marginRight": 16,
          }
        }
      >
        <Text
          allowFontScaling={false}
          dataSet={{}}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#E0E0E0",
                "fontSize": 24,
                "height": 24,
                "lineHeight": 24,
                "width": 24,
              },
              {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="reply"
        >
          󰑚
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
